google-cloud-bigquery/lib/google/cloud/bigquery/dataset.rb

```diff
@@ -1079,11 +1079,11 @@ def query query, params: nil, external: nil, max: nil, cache: true,
                       legacy_sql: legacy_sql, standard_sql: standard_sql,
                       params: params }
           options[:dataset] ||= self
-          updater = QueryJob::Updater.from_options query, options
+          updater = QueryJob::Updater.from_options service, query, options
```

Comment:
Shouldn't `query` be calling `query_job` instead of creating its own updater?

Reply:
I agree that would be better, but may not make this change in this PR.